src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -462,61 +462,53 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
                 elementNames: default(ImmutableArray<string>),
                 compilation: compilation,
                 diagnostics: diagnostics,
-                shouldCheckConstraints: false);
+                shouldCheckConstraints: false,
+                errorPositions: default(ImmutableArray<bool>));
         }
 
         private BoundTupleLiteral DeconstructionVariablesAsTuple(CSharpSyntaxNode syntax, ArrayBuilder<DeconstructionVariable> variables, DiagnosticBag diagnostics, bool hasErrors)
         {
-            bool inferNames = this.Compilation.LanguageVersion.InferTupleElementNames();
             int count = variables.Count;
             var valuesBuilder = ArrayBuilder<BoundExpression>.GetInstance(count);
             var typesBuilder = ArrayBuilder<TypeSymbol>.GetInstance(count);
             var locationsBuilder = ArrayBuilder<Location>.GetInstance(count);
-            var namesBuilder = inferNames ? ArrayBuilder<string>.GetInstance(count) : null;
+            var namesBuilder = ArrayBuilder<string>.GetInstance(count);
             foreach (var variable in variables)
             {
                 if (variable.HasNestedVariables)
                 {
                     var nestedTuple = DeconstructionVariablesAsTuple(variable.Syntax, variable.NestedVariables, diagnostics, hasErrors);
                     valuesBuilder.Add(nestedTuple);
                     typesBuilder.Add(nestedTuple.Type);
-                    if (inferNames)
-                    {
-                        namesBuilder.Add(null);
-                    }
+                    namesBuilder.Add(null);
                 }
                 else
                 {
                     var single = variable.Single;
                     valuesBuilder.Add(single);
                     typesBuilder.Add(single.Type);
-                    if (inferNames)
-                    {
-                        namesBuilder.Add(ExtractDeconstructResultElementName(single));
-                    }
+                    namesBuilder.Add(ExtractDeconstructResultElementName(single));
                 }
                 locationsBuilder.Add(variable.Syntax.Location);
             }
 
-            ImmutableArray<string> tupleNames;
-            if (inferNames)
-            {
-                var uniqueFieldNames = PooledHashSet<string>.GetInstance();
-                RemoveDuplicateInferredTupleNames(namesBuilder, uniqueFieldNames);
-                uniqueFieldNames.Free();
-                tupleNames = namesBuilder.ToImmutableAndFree();
-            }
-            else
-            {
-                tupleNames = default(ImmutableArray<string>);
-            }
+            var uniqueFieldNames = PooledHashSet<string>.GetInstance();
+            RemoveDuplicateInferredTupleNames(namesBuilder, uniqueFieldNames);
+            uniqueFieldNames.Free();
```

Comment:
Unfortunately, no. Two of the usages use a fresh hashset (because they can't have explicit names in those scenarios, for instance in deconstruction), but one needs a hashset that already has explicit names (that's in `BindTupleExpression`).